pkg/kubelet/kubelet.go

```diff
@@ -1385,15 +1383,30 @@ func (kl *Kubelet) initializeRuntimeDependentModules() {
 	// container log manager must start after container runtime is up to retrieve information from container runtime
 	// and inform container to reopen log file after log rotation.
 	kl.containerLogManager.Start()
+
 	// Adding Registration Callback function for CSI Driver
-	kl.pluginManager.AddHandler(pluginwatcherapi.CSIPlugin, plugincache.PluginHandler(csi.PluginHandler))
+	if err := kl.registerCSIPluginHandler(); err != nil {
+		kl.recorder.Eventf(kl.nodeRef, v1.EventTypeWarning, events.KubeletSetupFailed, err.Error())
+		klog.Fatalf("failed to register CSI Plugin Handler. err: %v", err)
```

Comment:
Under what circumstances would an error occur here? Adding a `Fatal` call in the kubelet startup path makes me nervous.